scripts/database.py

```diff
@@ -39,55 +49,70 @@ def get_paragraph(mongo_collection, doc_ids : list):
     if documents:
         for doc in documents:
             doc['_id'] = str(doc['_id'])
-            if doc['_id'] not in doc_ids and check_queue(doc['_id']):
+            if doc['_id'] not in doc_ids and doc['_id'] not in recent_ids and not check_user_email(doc, email):
                 update_queue(doc['_id'])
                 doc_ids.append(doc['_id'])
-                return doc, doc_ids
-    
+                return doc, doc_ids, recent_ids
+
     raise Exception('No documents found')
-    
+
+
+def check_user_email(doc, email):
+    for user in doc['labels']:
+        if user['email'] == email:
+            return True
+    return False
+
+
+def get_recent_ids():
+    collection = get_document_collection('paragraph_queue')
+    docs = collection.find(
+        {'date': {'$gt': datetime.now() - timedelta(hours=1)}}, {'_id': 1})
+    return [doc['_id'] for doc in docs]
+
+
 def update_queue(_id):
-    client = pymongo.MongoClient(os.getenv('MONGO_URI'))
-    db = client['sdg_text_corpora']
-    collection = db['paragraph_queue']
+    collection = get_document_collection('paragraph_queue')
 
     doc = collection.find_one({'_id': _id})
     if doc:
-        collection.replace_one({'_id': _id}, {'_id' : _id, 'date' : datetime.now()})
+        collection.replace_one(
+            {'_id': _id}, {'_id': _id, 'date': datetime.now()})
     else:
-        collection.insert_one({'_id' : _id, 'date' : datetime.now()})
-
-def check_queue(_id):
-    client = pymongo.MongoClient(os.getenv('MONGO_URI'))
-    db = client['sdg_text_corpora']
-    collection = db['paragraph_queue']
-    doc = collection.find_one({'_id' : _id})
-    
-    if doc:
-        if datetime.now() - doc['date'] >= timedelta(hours=1):
-            return True
-        return False
-    else:
-        return True
-        
+        collection.insert_one({'_id': _id, 'date': datetime.now()})
 
 
-def update_paragraph(collection, _id, labels): 
+def update_paragraph(_id, labels, email):
     _id = ObjectId(_id)
+    collection = get_document_collection('test')
```

Comment:
@Guille-lg, this function should not be parametrised to avoid code duplication. Use an env variable inside the function definition instead.
```suggestion
    collection = get_document_collection()
```